### PR TITLE
CORE-9775 Support extendable finalization result

### PR DESCRIPTION
**Pull request by owenstanford**
UTXO finalization methods have been changed to return a new FinalizationResult interface, this allows additional results (such as automatic token locking) to be returned in future versions of the API without a breaking change.

### CODE REVIEW

File: components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/FinalizationResultImpl.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package net.corda.ledger.utxo.flow.impl
+
+import net.corda.v5.ledger.utxo.FinalizationResult
+import net.corda.v5.ledger.utxo.transaction.UtxoSignedTransaction
+
+class FinalizationResultImpl(private val utxoSignedTransaction:UtxoSignedTransaction):FinalizationResult {
```
Comment (lankydan) [inline components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/FinalizationResultImpl.kt]: Reformat, missing spaces.

File: simulator/runtime/src/main/kotlin/net/corda/simulator/runtime/ledger/utxo/SimFinalizationResultImpl.kt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package net.corda.simulator.runtime.ledger.utxo
+
+import net.corda.v5.ledger.utxo.FinalizationResult
+import net.corda.v5.ledger.utxo.transaction.UtxoSignedTransaction
+
+class SimFinalizationResultImpl(private val utxoSignedTransaction:UtxoSignedTransaction):FinalizationResult {
```
Comment (lankydan) [inline simulator/runtime/src/main/kotlin/net/corda/simulator/runtime/ledger/utxo/SimFinalizationResultImpl.kt:6]: Reformat.

**Review by lankydan — DISMISSED** [review]
2 reformatting comments, you can do it when you update the api version.

**Review by blsemo — DISMISSED** [review]
Looks fine to me

**Review by blsemo — APPROVED** [review]
Re-approved

**Pull request merged**